library/tools/src/main/java/com/adgear/anoa/library/tools/runnable/AvroConcatenator.java

```diff
@@ -67,8 +68,13 @@ public void run() {
       Function<InputStream, DataFileStream<GenericRecord>> builder = Unchecked.function(
           stream -> new DataFileStream<>(stream, new GenericDatumReader<>(schema)));
 
-      try (DataFileWriter<GenericRecord> writer = new DataFileWriter<>(
-          new GenericDatumWriter<GenericRecord>(schema))
+      final DataFileWriter<GenericRecord> dfw = new DataFileWriter<>(
+        new GenericDatumWriter<>(schema)
+      );
+      dfw.setCodec(CodecFactory.deflateCodec(6));
+      dfw.setSyncInterval(1048576);
```

Comment:
This too?

Reply:
I see...

Maintaining this dependency can be challenging, not sure if there is a better solution though

Reply:
Maybe a comment with a pointer to the above link is a good start?